src/components/CVE-2022-36804/CVE-2022-36804.tsx

```diff
@@ -41,47 +62,136 @@ const CVE202236804 = () => {
             ...(values.repoKey ? ["-r", values.repoKey] : []),
             ...(values.cmd ? ["-c", values.cmd] : []),
             ...(values.session ? ["--session", values.session] : []),
         ];
 
-        try {
-            const command = new Command("python3", args);
-            const result = await command.execute();
-            setOutput(result.stdout); // Update output state with the command's stdout
-        } catch (error) {
-            setOutput(`Error executing command: ${error}`); // Handle any errors
-        }
-        setLoading(false);
+        CommandHelper.runCommandGetPidAndOutput("python3", args, handleProcessData, handleProcessTermination)
+            .then(({ output, pid }) => {
+                setOutput(output);
+                setPid(pid);
+            })
+            .catch((error) => {
+                setOutput(error.message);
+                setLoading(false);
+                setAllowSave(true);
+            });
+    };
+
+    /**
+     * Function to clear output and reset save status.
+     */
+    const clearOutput = useCallback(() => {
+        setOutput("");
+        setHasSaved(false);
+        setAllowSave(false);
+    }, [setOutput]);
+
+    /**
+     * Function to handle completion of saving output to a text file.
+     * Sets the hasSaved flag to true and disallows further saving.
+     */
+    const handleSaveComplete = () => {
+        setHasSaved(true);
+        setAllowSave(false);
     };
 
     return (
-        <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
-            <LoadingOverlay visible={loading} />
-            <Stack>
-                {UserGuide(title, description_userguide)}
-                <TextInput label={"Domain/IP Address"} required {...form.getInputProps("url")} />
-                <TextInput label={"Project Key"} {...form.getInputProps("projKey")} />
-                <TextInput label={"Repository Name"} {...form.getInputProps("repoKey")} />
-                <TextInput label={"Command to Execute"} {...form.getInputProps("cmd")} />
-                <TextInput label={"Session ID (for private repos)"} {...form.getInputProps("session")} />
-                <Button type={"submit"}>Exploit</Button>
-                <ConsoleWrapper output={output} hideClearButton={true} />
-            </Stack>
-        </form>
+        <RenderComponent
+            title={title}
+            description={description}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            {!loadingModal && (
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description}
+                    dependencies={dependencies}
+                ></InstallationModal>
+            )}
+            <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
+                <LoadingOverlay visible={loading} />
```

Comment:
I do not think you need this line anymore because it has been replaced by LoadingOverlayAndCancelButton. 
```suggestion
         
```